authentication/pipeline/compliance.py

```diff
@@ -5,19 +5,33 @@
 from django.conf import settings
 from django.core import mail
 from social_core.exceptions import AuthException
+from social_core.pipeline.partial import partial
 
 from authentication.exceptions import (
     UserExportBlockedException,
     UserTryAgainLaterException,
 )
 from compliance import api
-
+from compliance.constants import REASON_CODES_RETRYABLE
+from profiles.serializers import UserSerializer
 
 log = logging.getLogger()
 
 
+def set_user_session_data(strategy, user):
+    """
+    Save the user data in session.
+
+    Args:
+        strategy (social_django.strategy.DjangoStrategy): the strategy used to authenticate
+        user(User): the user to save to the session
+    """
+    strategy.request.session["COMPLIANCE_DATA"] = UserSerializer(user).data
```

Comment:
I don't think we should be storing this data on the user's session and writing it out into the html, it should be incorporated as part of the social auth api responses instead.

Here's how i'd do it:
- Keep the redirect as-is
- When the view loads, the frontend queries the /retry endpoint immediately with no data, which will cause it to get the same response but in JSON form.
- Update the /retry endpoint to return the value of `UserSerializer` in the `extra_data` field, similar to how the code handling `RequirePasswordException` does.
  - This may get a little tricky here, because we don't want this whole data structure coming back in the response for `RequirePasswordException` for security reasons. That might require moving `extra_data` into `PartialException` instead of having it in the serializer.

Reply:
@rhysyngsun Not sure I understand.  Keep which redirect as is?  When which view loads?  The `RequirePasswordException` handling [here](https://github.com/mitodl/bootcamp-ecommerce/blob/mb/compliance/authentication/serializers.py#L282-L287)?

Reply:
Sorry, I guess maybe I'm not understanding the code correctly, what's the purpose of `COMPLIANCE_DATA`? It looked like it prepopulates this data on app init, which I don't think we want to do. Instead, the auth APIs should return whatever the frontend needs in its response.

Reply:
@rhyngsun I couldn't figure out how to pass along the serialized user data to the `/retry` page via social auth responses that worked for both new users registering for the first time and existing users logging in that had previously failed/skipped export compliance.  The session variable was a way around that.  It sounds like you know how to do it with social auth but I wasn't sure which redirect/view you're referring to and not sure looking at the code exactly how the `RequirePasswordException` leads to a populated `extra_data` field.

Reply:
@rhysyngsun I couldn't figure out how to pass along the serialized user data to the `/retry` page via social auth responses that worked for both new users registering for the first time and existing users logging in that had previously failed/skipped export compliance.  The session variable was a way around that.  It sounds like you know how to do it with social auth but I wasn't sure which redirect/view you're referring to and not sure looking at the code exactly how the `RequirePasswordException` leads to a populated `extra_data` field.

Reply:
I think I get it now - Modify the compliance serializer so that `extra_data` == `UserSerializer(user).data`.  When the new compliance retry page initially loads, make a post call to `api/register/email/retry` which will return an auth response that should include that `extra_data` field.  Use it to prepopulate the form instead of the session data.  Does that sound right?

Reply:
yeah, exactly

Reply:
Thanks, sometimes I'm a little slow and need to sleep on things! 😁